pyfr/util.py

```diff
@@ -168,3 +169,24 @@ def match_paired_paren(delim, n=5):
     rgt = rf'\{close}{ocset}*?)*?'
 
     return lft*n + mid + rgt*n
+
+def file_index(basedir, basename):
```

Comment:
These two functions would probably be better combined as a generator.  The `.send` method can be used to pass the current time in to the generator (and it can handle the state for nout itself).

Reply:
```python
def file_index(basedir, basename):
    n = 0

    # See if the basename appears to depend on {n}
    if re.search('{n[^}]*}', basename):
        # Quote and substitute
        bn = re.escape(basename)
        bn = re.sub(r'\\{n[^}]*\\}', r'(\\s*\\d+\\s*)', bn)
        bn = re.sub(r'\\{t[^}]*\\}', r'(?:.*?)', bn) + '$'

        for f in os.listdir(basedir):
            if (m := re.match(bn, f)):
                n = max(n, int(m[1]) + 1)

    while True:
        t = yield
        yield os.path.join(basedir, basename.format(t=t, n=n))
        n += 1
```

Once the generator is up and running you'd do `g.send(intg.tcurr); fname = next(g)`